Cargo.toml

```diff
@@ -14,12 +14,14 @@ env_logger = "0.8.2"
 eth2_ssz = "0.1.2"
 eth2_ssz_derive = "0.1.0"
 futures = "0.3.13"
+get_if_addrs = "0.5.3"
```

Comment:
I'm worried that this hasn't been updated in some time. Some alternatives might be [`heim`](https://crates.io/crates/heim) or [`interfaces`](https://crates.io/crates/interfaces). I think `heim` is more mature.

Reply:
I got stuck on getting async working with `heim`. If you feel like pairing with me to fix it, that would be cool. Otherwise, I've got an implementation with `interfaces` ready to go in c2a564c.

Reply:
I'm fine with `interfaces` if you are, but I'm okay with pairing too. I think you'll want to do something like:

```rust
use futures_util::stream::TryStreamExt;

let ifaces: Result<Vec<_>, _> = heim::net::nic().try_collect().await;
```

I'd actually have to sit down and give it a go though.